analyses/cell-type-ewings/exploratory_analysis/06-tumor-cell-state-assignment.Rmd

```diff
@@ -0,0 +1,606 @@
+So here I am going to take just the tumor cells and re-assign clusters. 
+I'm using the same parameters we use for the whole sample, leiden, modularity, resolution of 0.5, and nearest neighbors of 20. 
+I think if we really wanted to be stringent we would evaluate clusters here and find the best parameters to use, but I think starting with these is good. 
+
+
+```{r}
+# get a data frame of all tumor cells and re assigned clusters 
+subcluster_df <- sce_list |>
+  purrr::imap(\(sce, id){
+    
+    # pull out the barcodes that are from tumor cells
+    tumor_cells <- tumor_cells_df |> 
+      dplyr::filter(library_id == id) |> 
+      dplyr::pull(barcodes)
+    
+    # filter the sce to only those barcodes
+    filtered_sce <- sce[ , tumor_cells]
+    
+    # grab pcs for clustering
+    pcs <- reducedDim(filtered_sce)
+    
+    # get the clusters 
+    clusters <- bluster::clusterRows(
+      pcs,
+      bluster::NNGraphParam(
+        k = 20,
+        type = "jaccard",
+        cluster.fun = "leiden",
+        cluster.args = list(objective_function = "modularity",
+                            resolution = 0.5)
+      )
+    )
+    
+    # combine into a data frame with barcodes
+    cluster_df <- data.frame(
+      barcodes = rownames(pcs),
+      sub_cluster = clusters
+    )
+  }) |> 
+  # get clusters for both samples in one data frame
+  dplyr::bind_rows(.id = "library_id")
+
+# add clusters to tumor cells df for plotting
+tumor_cells_df <- tumor_cells_df |> 
+  dplyr::left_join(subcluster_df, by = c("library_id", "barcodes"))
+```
+
+The first thing we will do is plot the cluster assignments on the UMAP. 
+
+```{r}
+library_ids |>
+  purrr::map(\(id){
+    tumor_cells_df |>
+      dplyr::filter(library_id == id) |>
+      ggplot(aes(x = UMAP1, y = UMAP2, color = sub_cluster)) +
+      geom_point(alpha = 0.5, size = 0.1) +
+      theme(
+        aspect.ratio = 1
+      ) +
+      labs(title = id) +
+      scale_color_brewer(palette = "Dark2") +
+      guides(color = guide_legend(override.aes = list(alpha = 1, size = 1.5)))
+  })
+```
+
+Now we will look at the marker gene expression across all clusters for each sample. 
+First using a density plot and then a heatmap. 
+
+```{r, fig.height=10}
+library_ids |> 
+  purrr::map(\(id){
+    
+    density_df <- tumor_cells_df |> 
+      dplyr::filter(library_id == id)
+    
+    marker_gene_cols |>
+    purrr::map(\(column){
+      plot_density(
+        density_df,
+        column,
+        "sub_cluster"
+      ) 
+    }) |>
+    patchwork::wrap_plots(ncol = 1) +
+      patchwork::plot_annotation(id)
+    
+  })
+```
+
+
+```{r}
+library_ids |> 
+  purrr::map(\(id){
+    
+    filtered_df <- tumor_cells_df |> 
+      dplyr::filter(library_id == id)
+    
+    full_celltype_heatmap(filtered_df, marker_gene_cols, "sub_cluster", plot_name = id)
+    
+  })
+
+```
+
+It looks like for both samples we have some clear division between EWS high and EWS low cells. 
```

Comment:
Hmm.. what do we make of the fact that it barely looks like `annotation` (which is the sub clusters - can you modify the heatmap code to get that labeled as "sub_cluster" and not as "annotation"?) is clustered on the 2nd heatmap? Of course, the hierachical clustering algorithm that the heatmap is using is decently different from the one used to cluster in the first place, but the total lack of relationship I find odd. I'm curious, what does the heatmap look like if we first arrange the data by subcluster, and then turn off x-axis clustering so that the subclusters indeed are grouped together?

Reply:
Ok, now I'm looking at the next version of this heatmap after you group into putative cell states, and I feel less concerned! The rainbow look, shall we say, was probably just brought on by some _very similar_ sub-clusters.